run.py

```
@@ -409,31 +411,80 @@ def run_diffusion_processsing(**args):
+                            RuntimeError("Phase encoding direction not specified for diffusion data.")
+                        pos = "EMPTY"
+                        neg = "EMPTY"
+                        gdcoeffs = "None"
+                        for dwi in dwis:
+                            if "-" in layout.get_metadata(dwi)["PhaseEncodingDirection"]:
+                                neg = dwi
+                            else:
+                                pos = dwi
+                        try:
+                            echospacing = layout.get_metadata(pos)["EffectiveEchoSpacing"] * 1000
+                            dwi_stage_dict = OrderedDict([("DiffusionPreprocessing", partial(run_diffusion_processsing,
+                                                                                             posData=pos,
+                                                                                             negData=neg,
+                                                                                             path=args.output_dir,
+                                                                                             subject="sub-%s" % subject_label,
+                                                                                             echospacing=echospacing,
+                                                                                             PEdir=PEdir,
+                                                                                             gdcoeffs="NONE",
+                                                                                             dwiname=dwiname,
+                                                                                             n_cpus=args.n_cpus))])
+                            for stage, stage_func in dwi_stage_dict.iteritems():
+                                if stage in args.stages:
+                                    stage_func()
+                        except:
```

Please replace this with a more specific catch (narrower in the amount of code it encomapses and types of exceptions it is suppose to catch). Right now it will catch all exceptions making debugging really hard.